agent/containers/images/image-test-script.sh

```diff
@@ -10,8 +10,39 @@ DISTRO=$distrovar
 read -p "Specify image tag to test: " tagvar
 TAG=$tagvar
 
-read -p "Specify redis host: " REDIS_HOST
-read -p "Specify redis port: " REDIS_PORT
+badhost=true
+while [ "$badhost" == true ]
+do
+    read -p "Specify redis host (or press 'ENTER' for default '$CONTROLLER'): " REDIS_HOST
+    if [ -z "$REDIS_HOST" ]
+    then
+        REDIS_HOST=$CONTROLLER
+    fi
+    ping_first=$(ping -c 1 -W 1 ${REDIS_HOST} 2>&1 | grep 'Name or service not known')
+    ping_second=$(ping -c 1 -W 1 ${REDIS_HOST} 2>&1 | grep '100% packet loss')
+    if [ -n "$ping_first" ] || [ -n "$ping_second" ]
+    then
+        printf -- "Host specified either does not exist or is unreachable, please try again:\n"
+    else
+        badhost=false
+    fi
```

Comment:
The whole loop is basically repeated, but this phrase in particular could easily be refactored out into a function to be reused to validate a hostname.